src/modules/csm/tree.py

```diff
@@ -30,12 +47,10 @@ def decode(cls, content: bytes) -> Self:
     def encode(self) -> bytes:
         """Convert the underlying StandardMerkleTree to a binary representation"""
 
-        def default(o):
-            if isinstance(o, bytes):
-                return f"0x{o.hex()}"
-            raise ValueError(f"Unexpected type for json encoding, got {repr(o)}")
+        return TreeJSONEncoder(indent=2).encode(self.dump()).encode()
 
-        return json.dumps(self.tree.dump(), indent=2, default=default).encode()
+    def dump(self) -> TreeDump:
+        return {**self.tree.dump(), "stateCID": self.state_cid}
```

Comment:
> StandardMerkleTree.load(json.loads(content))
Since json format changed can be any problems here ^

Reply:
+ to snake_case

Reply:
decode method should also decode state_cid

Reply:
I've added a test specifically to ensure backward compatibility.